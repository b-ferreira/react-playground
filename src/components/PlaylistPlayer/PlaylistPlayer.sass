.PlaylistPlayer
  width: 100%
  max-width: 300px
  max-height: 80px
  overflow: hidden
  display: flex
  justify-content: center

  iframe
    width: 100%
    height: 80px

  @include tablet
    max-width: 738px

  @include desktop
    max-width: 1320px
