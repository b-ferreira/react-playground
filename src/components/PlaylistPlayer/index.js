import React from 'react'
import PropTypes from 'prop-types'

const PlaylistPlayer = ({ uri }) => {
  return (
    <div className="PlaylistPlayer">
      {
        uri ? (
          <iframe title="Playlist Player"
                  src={`https://open.spotify.com/embed?uri=${uri}`}
                  allowtransparency="true" allow="encrypted-media">
          </iframe>
        ) : (
          <span className="simple-message">
            Hmmm... it seems that we're getting problems with this playlist... Sorry.
          </span>
        )
      }
    </div>
  )
}

PlaylistPlayer.displayName = 'PlaylistPlayer'
PlaylistPlayer.propTypes = {
  uri: PropTypes.string
}

export default PlaylistPlayer

