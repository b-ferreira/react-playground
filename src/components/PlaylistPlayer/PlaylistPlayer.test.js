import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import PlaylistPlayer from './index'

const uriMock = 'spotify:user:spotify:playlist:37i9dQZF1DWYtKpmml7moA'

it('renders correctly', () => {
  const tree = renderer
    .create(<PlaylistPlayer uri={uriMock} />)
    .toJSON()
    expect(tree).toMatchSnapshot()
})

it('renders correctly without uri', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <PlaylistPlayer />,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <PlaylistPlayer uri={uriMock} />,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})
