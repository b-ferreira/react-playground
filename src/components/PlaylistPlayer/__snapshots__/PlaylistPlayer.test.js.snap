// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="PlaylistPlayer"
>
  <iframe
    allow="encrypted-media"
    allowtransparency="true"
    src="https://open.spotify.com/embed?uri=spotify:user:spotify:playlist:37i9dQZF1DWYtKpmml7moA"
    title="Playlist Player"
  />
</div>
`;
